openmdao/core/tests/test_dyn_sizing.py

```diff
@@ -0,0 +1,730 @@
+
+
+# This is based on passing size information through the system shown below
+# in all tests C starts with the shape information
+
+# +-----+
+# |     |
+# |  A  +-----+
+# |     |     |
+# +-----+     |
+#         +---+---+
+#         |       |
+#         |   B   +-----+
+#         |       |     |
+#         +-------+     |
+#                   +---+----+
+#                   |        |
+#                   |   C    +-----+
+#                   |        |     |
+#                   +--------+ +---+---+
+#                              |       |
+#                              |   D   +----+
+#                              |       |    |
+#                              +-------+    |
+#                                        +--+----+
+#                                        |       |
+#                                        |   E   |
+#                                        |       |
+#                                        +-------+
+
+class B(om.ExplicitComponent):
+
+    def setup(self):
+        # Inputs
+        self.add_input('in', copy_shape='out')
+        self.add_output('out', shape_by_conn=True)
+
+    def compute(self, inputs, outputs):
+        outputs['out'] = inputs['in']
+
+
+class C(om.ExplicitComponent):
+
+    def setup(self):
+        # Inputs
+        self.add_input('in', shape=4)
+        self.add_output('out', shape=9)
+
+    def compute(self, inputs, outputs):
+        outputs['out'] = np.arange(9)
+
+
+class D(om.ExplicitComponent):
+
+    def setup(self):
+        # Inputs
+        self.add_input('in', shape_by_conn=True)
+        self.add_output('out', copy_shape='in')
+
+    def compute(self, inputs, outputs):
+        outputs['out'] = inputs['in']
+
+
+class E(om.ExplicitComponent):
+
+    def setup(self):
+        # Inputs
+        self.add_input('in', shape_by_conn=True)
+        self.add_output('out', copy_shape='in')
+
+    def compute(self, inputs, outputs):
+        print(inputs['in'])
+        outputs['out'] = inputs['in']
+
+
+class B_distrib(om.ExplicitComponent):
+    def initialize(self):
+        self.options['distributed'] = True
+
+    def setup(self):
+        # Inputs
+        self.add_input('in', copy_shape='out')
+        self.add_output('out', shape_by_conn=True)
+
+    def compute(self, inputs, outputs):
+        outputs['out'] = inputs['in']
+
+
+class C_distrib(om.ExplicitComponent):
+    def initialize(self):
+        self.options['distributed'] = True
+
+    def setup(self):
+        # Inputs
+        if self.comm.rank == 0:
+            self.add_input('in', shape=1, src_indices=np.arange(0,1, dtype=int))
+        elif self.comm.rank == 1:
+            self.add_input('in', shape=2, src_indices=np.arange(1,3, dtype=int))
+        else:
+            self.add_input('in', shape=0, src_indices=np.arange(3,3, dtype=int))
+
+        self.add_output('out', shape=3)
+
+    def compute(self, inputs, outputs):
+        outputs['out'] *= self.comm.rank
```

Comment:
this is strange, could you explain what's going on here (computing output based on itself)?

Reply:
I updated this so the output now depends upon the input.